Content.Server/Health/BodySystem/Surgery/SurgeryData/BiologicalSurgeryData.cs

```diff
@@ -11,81 +12,125 @@
-                    if (_skinRetracted) 
-                    {
-                        if (_targetOrgan != null && toolType == SurgeryToolType.VesselCompression)
-                            return RemoveOrganSurgery;
-                        if (toolType == SurgeryToolType.Incision) //_targetOrgan is potentially given a value by DisconnectOrganSurgery.
-                            return DisconnectOrganSurgery;
-                        else if (toolType == SurgeryToolType.Cauterization)
-                            return CautizerizeIncisionSurgery;
-                    }
-                    else
-                    {
-                        if (toolType == SurgeryToolType.Retraction)
-                            return RetractSkinSurgery;
-                        else if (toolType == SurgeryToolType.Cauterization)
-                            return CautizerizeIncisionSurgery;
-                    }
-                }
-                else
-                {
-                    if (toolType == SurgeryToolType.VesselCompression)
-                        return ClampVesselsSurgery;
-                    else if (toolType == SurgeryToolType.Cauterization)
-                        return CautizerizeIncisionSurgery;
-                }
+                return RemoveBodyPartSurgery;
             }
-            else
+            if (!_skinOpened) //Case: skin is normal.
             {
-                if (toolType == SurgeryToolType.Incision)
+                if (toolType == SurgeryType.Incision)
                     return OpenSkinSurgery;
             }
+            else if (_skinOpened && !_vesselsClamped) //Case: skin is opened, but not clamped.
+            {
+                if (toolType == SurgeryType.VesselCompression)
+                    return ClampVesselsSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
+            else if (_skinOpened && _vesselsClamped && !_skinRetracted) //Case: skin is opened and clamped, but not retracted.
+            {
+                if (toolType == SurgeryType.Retraction)
+                    return RetractSkinSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
+            else if (_skinOpened && _vesselsClamped && _skinRetracted) //Case: skin is fully open.
+            {
+                if (_parent.Mechanisms.Count > 0 && toolType == SurgeryType.VesselCompression && (_disconnectedOrgans.Except(_parent.Mechanisms).Count() != 0 || _parent.Mechanisms.Except(_disconnectedOrgans).Count() != 0))
+                    return LoosenOrganSurgery;
+                else if (_disconnectedOrgans.Count > 0 && toolType == SurgeryType.Incision)
+                    return RemoveOrganSurgery;
+                else if (toolType == SurgeryType.Cauterization)
+                    return CautizerizeIncisionSurgery;
+            }
```

Comment:
these 3 else ifs with a //Case:
all require `_skinOpened`, but since they're all else ifs to an `if(!_skinOpened)`, that's already true.

You can remove `_skinOpened &&` from all 3 else ifs

However, for readability, I'd rearrange the entire thing to:

```csharp
if(_skinOpened)
{
    if(_vesselsClamped)
    {
        if(_skinRetracted)
        {
            if (_parent.Mechanisms.Count > 0 && toolType == SurgeryType.VesselCompression 
                && (_disconnectedOrgans.Except(_parent.Mechanisms).Count() != 0 
                || _parent.Mechanisms.Except(_disconnectedOrgans).Count() != 0))
                return LoosenOrganSurgery;
            else if (_disconnectedOrgans.Count > 0 && toolType == SurgeryType.Incision)
                return RemoveOrganSurgery;
            else if (toolType == SurgeryType.Cauterization)
                return CautizerizeIncisionSurgery;
        } 
        else 
        {
            if (toolType == SurgeryType.Retraction)
                return RetractSkinSurgery;
            else if (toolType == SurgeryType.Cauterization)
                return CautizerizeIncisionSurgery;
        }
    } 
    else 
    {
        if (toolType == SurgeryType.VesselCompression)
            return ClampVesselsSurgery;
        else if (toolType == SurgeryType.Cauterization)
            return CautizerizeIncisionSurgery;
    }
} 
else 
{
    if(toolType == SurgeryType.Incision)
        return OpenSkinSurgery;
}
```

Reply:
I think mine is more readable since it has the surgery steps in order - i.e., first you open the skin, then you retract it, etc etc.
But you're right about the redundant checks. I removed them.

Reply:
Mine's in order too, just reversed :P